### ISSUE TITLE
ansible failing when trying to provision n1-standard-1 GCE instance

### ISSUE THREAD

**Issue by lucasduffey** [trailofbits/algo#616]
### about
I am running `algo` and trying to set up a vpn. Modified config.cfg and changed the following line (it worked with initial f1-micro):
```
  gce:
    size: n1-standard-1 # f1-micro was original value
    image: ubuntu-1604
```

### OS / Environment
Ubuntu 16 LTS

### error log
```
$ ./algo 

  What provider would you like to use?
    1. DigitalOcean
    2. Amazon EC2
    3. Microsoft Azure
    4. Google Compute Engine (only for testing, see issue #369)
    5. Install to existing Ubuntu 16.04 server

Enter the number of your desired provider
: 4

Enter the local path to your credentials JSON file (https://support.google.com/cloud/answer/6158849?hl=en&ref_topic=6262490#serviceaccounts):
: [VALID_JSON_FILE_WAS_HERE]


Name the vpn server:
[algo]: 


  What zone should the server be located in?
    1. Western US       (Oregon A)
    2. Western US       (Oregon B)
    3. Western US       (Oregon C)
    4. Central US       (Iowa A)
    5. Central US       (Iowa B)
    6. Central US       (Iowa C)
    7. Central US       (Iowa F)
    8. Eastern US       (Northern Virginia A)
    9. Eastern US       (Northern Virginia B)
    10. Eastern US      (Northern Virginia C)
    11. Eastern US      (South Carolina B)
    12. Eastern US      (South Carolina C)
    13. Eastern US      (South Carolina D)
    14. Western Europe  (Belgium B)
    15. Western Europe  (Belgium C)
    16. Western Europe  (Belgium D)
    17. Southeast Asia  (Singapore A)
    18. Southeast Asia  (Singapore B)
    19. East Asia       (Taiwan A)
    20. East Asia       (Taiwan B)
    21. East Asia       (Taiwan C)
    22. Northeast Asia  (Tokyo A)
    23. Northeast Asia  (Tokyo B)
    24. Northeast Asia  (Tokyo C)
Please choose the number of your zone. Press enter for default (#14) zone.
[14]: 

Do you want macOS/iOS clients to enable "VPN On Demand" when connected to cellular networks?
[y/N]: 

Do you want macOS/iOS clients to enable "VPN On Demand" when connected to Wi-Fi?
[y/N]: 

Do you want to install a DNS resolver on this VPN server, to block ads while surfing?
[y/N]: 

Do you want each user to have their own account for SSH tunneling?
[y/N]: 

Do you want to apply operating system security enhancements on the server? (warning: replaces your sshd_config)
[y/N]: 

Do you want the VPN to support Windows 10 or Linux Desktop clients? (enables compatible ciphers and key exchange, less secure)
[y/N]: 

Do you want to retain the CA key? (required to add users in the future, but less secure)
[y/N]: 
 _____________________________
< PLAY [Configure the server] >
 -----------------------------
        \   ^__^
         \  (oo)\_______
            (__)\       )\/\
                ||----w |
                ||     ||

 ______________
< TASK [setup] >
 --------------
        \   ^__^
         \  (oo)\_______
            (__)\       )\/\
                ||----w |
                ||     ||

ok: [localhost]
 _____________________________________
< TASK [Generate the SSH private key] >
 -------------------------------------
        \   ^__^
         \  (oo)\_______
            (__)\       )\/\
                ||----w |
                ||     ||

ok: [localhost]
 ____________________________________
< TASK [Generate the SSH public key] >
 ------------------------------------
        \   ^__^
         \  (oo)\_______
            (__)\       )\/\
                ||----w |
                ||     ||

ok: [localhost]
 ____________________________________________
< TASK [Change mode for the SSH private key] >
 --------------------------------------------
        \   ^__^
         \  (oo)\_______
            (__)\       )\/\
                ||----w |
                ||     ||

ok: [localhost]
 ____________________________________________
< TASK [Ensure the dynamic inventory exists] >
 --------------------------------------------
        \   ^__^
         \  (oo)\_______
            (__)\       )\/\
                ||----w |
                ||     ||

ok: [localhost]
 _____________________________
< TASK [cloud-gce : set_fact] >
 -----------------------------
        \   ^__^
         \  (oo)\_______
            (__)\       )\/\
                ||----w |
                ||     ||

ok: [localhost]
 _____________________________
< TASK [cloud-gce : set_fact] >
 -----------------------------
        \   ^__^
         \  (oo)\_______
            (__)\       )\/\
                ||----w |
                ||     ||

ok: [localhost]
 _____________________________
< TASK [cloud-gce : set_fact] >
 -----------------------------
        \   ^__^
         \  (oo)\_______
            (__)\       )\/\
                ||----w |
                ||     ||

ok: [localhost]
 _______________________________________
< TASK [cloud-gce : Network configured] >
 ---------------------------------------
        \   ^__^
         \  (oo)\_______
            (__)\       )\/\
                ||----w |
                ||     ||

fatal: [localhost]: FAILED! => {"changed": false, "failed": true, "msg": "Unexpected response: ('ports'). Detail: Traceback (most recent call last):\n  File \"/tmp/ansible_Hn0xl2/ansible_module_gce_net.py\", line 352, in main\n    if allowed_list and (sorted_allowed_list(allowed_list) != sorted_allowed_list(fw.allowed)):\n  File \"/tmp/ansible_Hn0xl2/ansible_module_gce_net.py\", line 230, in sorted_allowed_list\n    return sorted(allowed_by_protocol, key=lambda y: y['ports'].sort())\n  File \"/tmp/ansible_Hn0xl2/ansible_module_gce_net.py\", line 230, in <lambda>\n    return sorted(allowed_by_protocol, key=lambda y: y['ports'].sort())\nKeyError: 'ports'\n"}
 __________________________
< TASK [cloud-gce : debug] >
 --------------------------
        \   ^__^
         \  (oo)\_______
            (__)\       )\/\
                ||----w |
                ||     ||

ok: [localhost] => {
    "fail_hint": [
        "Sorry, but something went wrong!", 
        "Please check the troubleshooting guide.", 
        "https://trailofbits.github.io/algo/troubleshooting.html"
    ]
}
 _________________________
< TASK [cloud-gce : fail] >
 -------------------------
        \   ^__^
         \  (oo)\_______
            (__)\       )\/\
                ||----w |
                ||     ||

fatal: [localhost]: FAILED! => {"changed": false, "failed": true, "msg": "Failed as requested from task"}
 ____________
< PLAY RECAP >
 ------------
        \   ^__^
         \  (oo)\_______
            (__)\       )\/\
                ||----w |
                ||     ||

localhost                  : ok=9    changed=0    unreachable=0    failed=1   


```

**Comment by lucasduffey** [comment]
during the `cloud-gce : Network configured` task it gets the following error:

```
 FAILED! => {"changed": false, "failed": true, "msg": "Unexpected response: ('ports'). Detail: Traceback (most recent call last):\n  File \"/tmp/ansible_Hn0xl2/ansible_module_gce_net.py\", line 352, in main\n    if allowed_list and (sorted_allowed_list(allowed_list) != sorted_allowed_list(fw.allowed)):\n  File \"/tmp/ansible_Hn0xl2/ansible_module_gce_net.py\", line 230, in sorted_allowed_list\n    return sorted(allowed_by_protocol, key=lambda y: y['ports'].sort())\n  File \"/tmp/ansible_Hn0xl2/ansible_module_gce_net.py\", line 230, in <lambda>\n    return sorted(allowed_by_protocol, key=lambda y: y['ports'].sort())\nKeyError: 'ports'\n"}
```

It looks like "ports" isn't a valid key.

**Comment by gunph1ld** [comment]
Related to [this ansible bug](https://github.com/ansible/ansible/issues/20506). fixed in Ansible 2.3
